.github/workflows/release-proxy-init.yml

```diff
@@ -0,0 +1,96 @@
+name: Proxy-init release
+
+on:
+  pull_request:
+    paths:
+      - .github/workflows/release-proxy-init.yml
```

Comment:
All jobs will run when this file changes, right? Do we want that? We don't do this in `release.yml` for linkerd2.

If we do want this to run to make sure it still works properly, should we make some of the jobs conditional on it being a tag that was pushed?